app.js

```diff
@@ -1,8 +1,168 @@
 // import functions and grab DOM elements
+const goblinsVanquishedEl = document.querySelector('#goblins-vanquished');
+
+const playerAvatarEl = document.querySelector('#player-avatar');
+const playerStatsEl = document.querySelector('#player-stats');
+const playerHpSpanEl = document.querySelector('span');
+
+const matchupSectionEl = document.querySelector('#matchup-section');
+const matchupMessageEl = document.querySelector('#matchup-message');
+
+const goblinsSectionEl = document.querySelector('#goblins-section');
+const addGoblinFormEl = document.querySelector('#add-goblin-form');
+const goblinsDivEl = document.querySelector('#goblins-div');
+
 
 // let state
+let goblins = [
+    {
+        name: 'Bejorkus',
+        emoji: '👹',
+        hp: 4
+    },
+    {
+        name: 'Groomf',
+        emoji: '👹',
+        hp: 5
+    },
+    {
+        name: 'Horseclaw',
+        emoji: '👹',
+        hp: 3
+    },
+    {
+        name: 'Steve',
+        emoji: '👹',
+        hp: 1
+    }
+];
+
+let numberOfGoblinsVanquished = 0;
+
+let playerHealth = 10;
 
 // set event listeners 
-  // get user input
-  // use user input to update state 
-  // update DOM to reflect the new state
+addGoblinFormEl.addEventListener('submit', (e) => {
+    e.preventDefault();
+
+    const data = new FormData(addGoblinFormEl);
+
+    // get user input
+    const userGoblinName = data.get('goblin-name-input');
+
+    // use user input to update state
+    const newGoblin = {
+        name: userGoblinName,
+        emoji: '👹',
+        hp: Math.ceil(Math.random() * 6),
+    };
+
+    goblins.push(newGoblin);
+
+    addGoblinFormEl.reset();
+    // update DOM to reflect the new state
+    displayGoblins();
+});
+
+function displayGoblins() {
+    goblinsDivEl.textContent = '';
+    
+    for (let goblin of goblins) {
+  
+        const goblinEl = document.createElement('div');
+        goblinEl.textContent = `${goblin.name} ${goblin.emoji} ${goblin.hp} hp`;
+      
+        goblinEl.classList.add('goblin');
+      
+        goblinsDivEl.append(goblinEl);
```

Comment:
Can you refactor this to call a pure function called `renderGoblin` that takes a single goblin object as an argument and returns this goblinEl that you're creating here?

```js
const goblinEl = renderGoblin(goblin);
goblinsDivEl.append(goblinEl);
```

Reply:
Got it now. Thanks!